File: Ejercicios/5. Duck/Java/Before/src/main/duck/Duck.java
package duck;

public abstract class Duck {
    public Object fly()
    {
        return DuckMessages.Fly;
    }

    public Object quack()
    {
        return DuckMessages.Quack;
    }

    public String swim()
    {
        return DuckMessages.Swim;
    }

    public abstract Object display();
}

File: Ejercicios/5. Duck/Net/Before/Duck/Duck.cs
namespace Duck
{
    public abstract class Duck
    {
        public virtual object Fly()
        {
            return DuckMessages.Fly;
        }

        public virtual object Quack()
        {
            return DuckMessages.Quack;
        }

        public string Swim()
        {
            return DuckMessages.Swim;
        }

        public abstract object Display();
    }
}